logic/transactions/pool.js

```diff
@@ -0,0 +1,922 @@
+		if (receiptTransactions.ready.length > 0) {
+			receiptTransactions.ready.forEach(function (receiptTransaction) {
+				if (receiptTransaction.type === transactionTypes.SEND) {
+					poolBalance = poolBalance.plus(receiptTransaction.amount.toString());
+				}
+			});
+		}
+
+		// total balance
+		var balance = new bignum(account.balance.toString());
+		balance = balance.plus(poolBalance);
+
+		// Check confirmed sender balance
+		var amount = new bignum(transaction.amount.toString()).plus(transaction.fee.toString());
+		var exceeded = balance.lessThan(amount);
+
+		if (exceeded) {
+			return setImmediate(cb, [
+				'Account does not have enough LSK:', sender.address,
+				'balance:', balance.div(Math.pow(10,8))
+			].join(' '));
+		}
+
+		return setImmediate(cb, null, [
+			'balance:', balance.div(Math.pow(10,8))
+		].join(' '));
+	});
+};
+
+/**
+ * Validates a transaction and adds it to the verified pool list.
+ * @implements {__private.addToVerified}
+ * @param {transaction} transaction
+ * @param {boolean} broadcast
+ * @param {function} cb - Callback function.
+ * @return {setImmediateCallback} error | cb
+ */
+TransactionPool.prototype.addFromPublic = function (transaction, broadcast, cb) {
+	__private.addToVerified(transaction, broadcast, cb);
+};
+
+/**
+ * Adds transactions to unverified pool list.
+ * @implements {__private.addToUnverified}
+ * @param {transaction} transactions
+ * @param {boolean} broadcast
+ * @param {function} cb - Callback function.
+ * @return {setImmediateCallback} error | cb
+ */
+TransactionPool.prototype.addFromPeer = function (transactions, broadcast, cb) {
+	if (!Array.isArray(transactions)) {
+		transactions = [transactions];
+	}
+	async.eachSeries(transactions, function (transaction, cb) {
+		__private.addToUnverified(transaction, broadcast, cb);
+	}, function (err) {
+		return setImmediate(cb, err);
+	});
+};
+
+/**
+ * Adds transactions to verified.ready pool list.
+ * @implements {__private.addToPoolList}
+ * @implements {delete}
+ * @param {transaction} transactions
+ * @param {function} cb - Callback function.
+ * @return {setImmediateCallback} error | cb
+ */
+TransactionPool.prototype.addReady = function (transactions, cb) {
+	if (!Array.isArray(transactions)) {
+		transactions = [transactions];
+	}
+	var resetReceivedAt = new Date();
+	async.eachSeries(transactions, function (transaction, eachSeriesCb) {
+		self.delete(transaction);
+		transaction.receivedAt = resetReceivedAt;
+		__private.addToPoolList(transaction, pool.verified.ready, eachSeriesCb);
+	}, function (err) {
+		return setImmediate(cb, err);
+	});
+};
+
+/**
+ * Creates and adds signature to multisignature transaction.
+ * @implements {__private.createSignature}
+ * @param {String} transactionId transaction id
+ * @param {String} secret secret
+ * @param {function} cb - Callback function.
+ * @return {setImmediateCallback} error | cb
+ */
+TransactionPool.prototype.addSignature = function (transactionId, secret, cb) {
```

Comment:
Should it be part of tx pool?

Reply:
Transaction pool knows pending multisignatures, if we keep this logic here, we won't need other components to handle signature transactions